lib/peopleHR.ts

```diff
@@ -71,6 +72,81 @@ async function parseXml(xmlString: string): Promise<any> {
     );
 }
 
+/**
+ * Pulls the PeopleHR RSS feed data from the cache,
+ * or fetches it from the feed if the cache has expired
+ * @returns JSON of the RSS feed
+ */
+async function getJobPostingData() {
+    const cacheFilePath = path.join(
+        process.cwd(),
+        'files',
+        'peopleHRcache.json',
+    );
+
+    if (existsSync(cacheFilePath)) {
+        const cacheFile = readFileSync(cacheFilePath, 'utf8');
+        const cacheJSON = JSON.parse(cacheFile);
+        const cacheTTL = 10 * 1000; //One hour 60 * 60 * 1000
+
+        if (cacheJSON.lastUpdated + cacheTTL < Date.now()) {
+            // The cache has timed out, fetch new data from PeopleHR
+
+            const peopleHRJobPostingsXML = await fetchPeopleHRFeed();
+
+            if (peopleHRJobPostingsXML === null) {
+                // In the event of a server error, don't update the job listings
+                // if we have cached listings still available to use.
+
+                const fileData = {
+                    lastUpdated: Date.now(),
+                    jobJSON: cacheJSON.jobJSON,
+                };
+
+                writeFileSync(cacheFilePath, JSON.stringify(fileData));
```

Comment:
**Suggestion**: again some more try catch statements would be nice for operations like these, perhaps making sure they will be compatiable with Sentry.

try {
    writeFileSync(cacheFilePath, JSON.stringify(fileData));
} catch (e) {
      console.error('Error writing peopleHR jobs to cache file: ', e)
}